### PR TITLE
:hammer: clean up

### PR DESCRIPTION
**Pull request by kartik4949**
making pooling ops helper generic

### CODE REVIEW

File: tinygrad/ops.py [hunk 1]
```diff
@@ -224,8 +225,7 @@ def backward(ctx, grad_output):
 
     # needs to be optimized
     gdx = np.zeros((bs,ctx.groups,cin,OY,OX), dtype=tx.dtype)
-    for Y in range(grad_output.shape[2]):
-      for X in range(grad_output.shape[3]):
+    for Y, X in itertools.product(range(grad_output.shape[2]),range(grad_output.shape[3])):
```
Comment (geohot) [inline tinygrad/ops.py:228]: Think the left side is more readable

File: tinygrad/ops.py [hunk 2]
```diff
@@ -238,54 +238,49 @@ def backward(ctx, grad_output):
...
 
   @staticmethod
   def backward(ctx, grad_output):
     idxs,s = ctx.saved_tensors
-    return unstack_for_pool(
-      lambda idx: grad_output * (idxs == idx),
-      s, *ctx.kernel_size)
+    return pool(s, *ctx.kernel_size,
+      fxn = lambda idx: grad_output * (idxs == idx)
+      )
```
Comment (geohot) [inline tinygrad/ops.py:268]: This paren can go on the above line.

File: tinygrad/ops.py (hunk 1 above)
Comment (kartik4949) [reply tinygrad/ops.py:228]: i was writing this in context of reduction of lines

Comment (kartik4949) [reply tinygrad/ops.py:228]: i can remove it though :)

File: tinygrad/ops.py (hunk 2 above)
Comment (kartik4949) [reply tinygrad/ops.py:268]: idk what u mean by this.

Comment (geohot) [reply tinygrad/ops.py:268]: fxn = lambda idx: grad_output * (idxs == idx))

File: tinygrad/ops.py (hunk 1 above)
Comment (kartik4949) [reply tinygrad/ops.py:228]: sure!

**Comment by geohot** [comment]
Will merge after itertools product is gone.

File: tinygrad/ops.py [hunk 3]
```diff
@@ -238,54 +239,45 @@ def backward(ctx, grad_output):
...
-  my, mx = (x.shape[2]//py)*py, (x.shape[3]//px)*px
-  stack = []
-  xup = x[:, :, :my, :mx]
-  for Y in range(py):
-    for X in range(px):
-      stack.append(xup[:, :, Y::py, X::px][None])
-  return np.concatenate(stack, axis=0)
-
-def unstack_for_pool(fxn, s, py, px):
+def pool(x, py, px, fxn=None):
```
Comment (geohot) [inline tinygrad/ops.py:242]: This really shouldn't be one function. The codepath is totally different depending on fxn.

The goal is not less lines. The goal is readable code first and foremost, then less lines. This is hard to read.

Comment (kartik4949) [reply tinygrad/ops.py:242]: alrite, will delete pr

**Pull request closed without merging**